ai2thor/tests/test_unity.py

```diff
@@ -110,18 +124,54 @@ def test_rotate_left():
     assert e.metadata['agent']['rotation']['z'] == 0.0
 
+
 def test_add_third_party_camera():
 
-    assert len(controller.last_event.metadata['thirdPartyCameras']) == 0
-    e = controller.step(dict(action='AddThirdPartyCamera', position=dict(x=1.2, y=2.3, z=3.4), rotation=dict(x=30, y=40,z=50)))
-    assert len(e.metadata['thirdPartyCameras']) == 1
-    assert_near(e.metadata['thirdPartyCameras'][0]['position'], dict(x=1.2, y=2.3, z=3.4))
-    assert_near(e.metadata['thirdPartyCameras'][0]['rotation'], dict(x=30, y=40, z=50))
-    assert len(e.third_party_camera_frames) == 1
-    assert e.third_party_camera_frames[0].shape == (300,300,3)
-    e = controller.step(dict(action='UpdateThirdPartyCamera', thirdPartyCameraId=0, position=dict(x=2.2, y=3.3, z=4.4), rotation=dict(x=10, y=20,z=30)))
-    assert_near(e.metadata['thirdPartyCameras'][0]['position'], dict(x=2.2, y=3.3, z=4.4))
-    assert_near(e.metadata['thirdPartyCameras'][0]['rotation'], dict(x=10, y=20, z=30))
+    expectedPosition = dict(x=1.2, y=2.3, z=3.4)
+    expectedRotation = dict(x=30, y=40, z=50)
+    expectedFieldOfView = 45.0
+    assert len(controller.last_event.metadata[MultiAgentMetadata.thirdPartyCameras]) == 0, 'there should be 0 cameras'
+
+    e = controller.step(dict(action=Actions.AddThirdPartyCamera, position=expectedPosition, rotation=expectedRotation, fieldOfView=expectedFieldOfView))
+    assert len(e.metadata[MultiAgentMetadata.thirdPartyCameras]) == 1, 'there should be 1 camera'
+    camera = e.metadata[MultiAgentMetadata.thirdPartyCameras][0]
+    assert_near(camera[ThirdPartyCameraMetadata.position], expectedPosition, 'initial position should have been set')
+    assert_near(camera[ThirdPartyCameraMetadata.rotation], expectedRotation, 'initial rotation should have been set')
+    assert camera[ThirdPartyCameraMetadata.fieldOfView] == expectedFieldOfView, 'initial fieldOfView should have been set'
+
+
+def test_update_third_party_camera():
+
+    expectedPosition = dict(x=2.2, y=3.3, z=4.4)
+    expectedRotation = dict(x=10, y=20, z=30)
+    expectedInitialFieldOfView = 45.0
+    expectedFieldOfView2 = 55.0
+    expectedFieldOfViewDefault = 90.0
+    assert len(controller.last_event.metadata[MultiAgentMetadata.thirdPartyCameras]) == 1, 'there should be 1 camera'
+
+    e = controller.step(dict(action=Actions.UpdateThirdPartyCamera, thirdPartyCameraId=0, position=expectedPosition, rotation=expectedRotation))
+    camera = e.metadata[MultiAgentMetadata.thirdPartyCameras][0]
+    assert_near(camera[ThirdPartyCameraMetadata.position], expectedPosition, 'position should have been updated')
+    assert_near(camera[ThirdPartyCameraMetadata.rotation], expectedRotation, 'rotation should have been updated')
+    assert camera[ThirdPartyCameraMetadata.fieldOfView] == expectedInitialFieldOfView, 'fieldOfView should not have changed'
+
+    # 0 is a special case, since nullable float does not get encoded properly, we need to pass 0 as null
```

Comment:
note, it would be preferable to allow serializing nullables, but it looks like nullables dont get passed... so i need to reserve 0 as null.

Reply:
you can make the `fieldOfView` a `nullable` in C#, using `?` and then if you don't pass anything or pass `None`, you can detect the nulled variable and have a special case for it. See `gridColor` in `ServerAction`.

Reply:
i tried that, and when i did, the c# return object (`ThirdPartyCameraMetadata` in AgentManager.cs)
stopped returning a fieldOfView even when I was setting it... reverting to not use nullable fixed the issue. I would like this to work as you say, but i would need some help understanding why the serialized ThirdPartyCameraMetadata that is returned is missing the field even when it is not null.